### ISSUE TITLE
Sentinel instance is not assignable to `Sentinel` type

### ISSUE THREAD

**Issue by sobolevn** [microsoft/pyright#11529]
When I write:

```python
from typing_extensions import Sentinel 

X = Sentinel("X")

def inspect_sentinel(arg: Sentinel) -> None:
    ...

inspect_sentinel(arg=X)
```

I get: 

```
Argument of type "X" cannot be assigned to parameter "arg" of type "Sentinel" in function "inspect_sentinel"
  "X" is not assignable to "Sentinel"  (reportArgumentType)
```

But, `X` is an instance of a `Sentinel` type. It should be possible to pass any sentinel where `Sentinel` type is expected. For example: for type inspection, as I do here.

Playground: https://pyright-play.net/?strict=true&locale=en&code=GYJw9gtgBALgngBwJYDsDmB9ApgDxllAZyTCKiQgTBBigGUCZUsAbKAKHYA0oBeexsxYAKAERdRASk4ATLMHJEEWAMYwMhQSlbCAhiDQAuASibaWkqAFoAfFABypLIfZQ3UAHRfOqQsrUaWjr6aLxckkA

**Comment by bram-tv** [comment]
It appears this got broken in pyright v1.1.410;
Extending your example slightly with `reveal_type` added:
```python
from typing_extensions import Sentinel

X = Sentinel("X")


def inspect_sentinel(arg: Sentinel) -> None: ...


reveal_type(X)
reveal_type(inspect_sentinel)
inspect_sentinel(arg=X)
```

Output with pyright v1.1.409:
```
.../gh-11529.py
  .../gh-11529.py:9:13 - information: Type of "X" is "Sentinel"
  .../gh-11529.py:10:13 - information: Type of "inspect_sentinel" is "(arg: Sentinel) -> None"
0 errors, 0 warnings, 2 informations
```

Output with pyright v1.1.410 (and v1.1.411):
```
.../gh-11529.py
  .../gh-11529.py:9:13 - information: Type of "X" is "X"
  .../gh-11529.py:10:13 - information: Type of "inspect_sentinel" is "(arg: Sentinel) -> None"
  .../gh-11529.py:11:22 - error: Argument of type "X" cannot be assigned to parameter "arg" of type "Sentinel" in function "inspect_sentinel"
    "X" is not assignable to "Sentinel" (reportArgumentType)
1 error, 0 warnings, 2 informations
```

Looking at the list of changes https://github.com/microsoft/pyright/releases/tag/1.1.410 there is not an obvious commit that triggered it.
My best guess it's somewhere in the [Push pylance changes to pyright](https://github.com/microsoft/pyright/pull/11396)...
(In my opinion that is not a very useful message or even commit, it includes too much in a single commit to be able to narrow down why the behavior changed)


For completness: output of mypy v1.20.2 and mypy v2.1.0:
```
gh-11529.py:9: note: Revealed type is "typing_extensions.Sentinel"
gh-11529.py:10: note: Revealed type is "def (arg: typing_extensions.Sentinel)"
Success: no issues found in 1 source file
```

**Comment by bram-tv** [comment]
I guess this change was made to conform to [PEP 661](https://peps.python.org/pep-0661/).

Its example contains:
```python
MISSING = sentinel('MISSING')

def foo(value: int | MISSING = MISSING) -> int:
    ...
```
Note that it uses: `int | MISSING` and not `int | sentinel`.

Possible related discussion of typing extensions: https://github.com/python/typing_extensions/issues/742

So I guess the work-around for now is to use something like:
```python
def inspect_sentinel(arg: X | Sentinel) -> None:
```

**Comment by edgarrmondragon** [comment]
Note that mypy in `master` (2.4.0-dev) produces the expected result:

```
gh-11529.py:9: note: Revealed type is "X?"
gh-11529.py:10: note: Revealed type is "def (arg: sentinel)"
```
